src/main/java/org/traccar/protocol/ArknavProtocolDecoder.java

```diff
@@ -59,23 +62,45 @@ protected Object decode(
             return null;
         }
 
-        Position position = new Position(getProtocolName());
-
         DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
         if (deviceSession == null) {
             return null;
         }
-        position.setDeviceId(deviceSession.getDeviceId());
 
-        position.setValid(parser.next().equals("A"));
-        position.setLatitude(parser.nextCoordinate());
-        position.setLongitude(parser.nextCoordinate());
-        position.setSpeed(parser.nextDouble(0));
+        Position position = new Position(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+        long status = parser.nextHexLong();
+        String model = parser.next();
+        Boolean valid_geo = parser.next().equals("A");
+        position.setValid(valid_geo);            
+        double latitude = parser.nextCoordinate();
+        double longitude = parser.nextCoordinate();
+        if (valid_geo || (latitude != 0 && longitude != 0)) {
+            position.setLatitude(latitude);
+            position.setLongitude(longitude);
+        }
+        position.setSpeed(UnitsConverter.kphFromKnots(parser.nextDouble(0)));
```

Comment:
Why are you doing this? This makes no sense to me. And it also breaks naming convention.

Reply:
I have changed back the code.

Reply:
I found in the documentation that these parameters can be configured in the configuration file.
No more changes to the code